contracts/Trader.sol

```diff
@@ -112,39 +82,51 @@ contract Trader {
      * @return the specified order
      * @dev Performs its own bounds check on the array access
      */
-    function grabOrder(
-        Types.SignedLimitOrder[] memory orders,
-        uint256 index
-    ) internal view returns (Types.LimitOrder memory) {
-            require(index <= orders.length, "TDR: Out of bounds access");
-
-            Types.SignedLimitOrder memory currentSigned = orders[index];
-            
-            /* verify signature and nonce */
-            require(
-                verify(
-                    currentSigned.order.user,
-                    currentSigned.order,
-                    currentSigned.sigR,
-                    currentSigned.sigS,
-                    currentSigned.sigV
-                ),
-                "TDR: incorrect order sig or nonce"
-            );
+    function grabOrder(Types.SignedLimitOrder[] memory orders, uint256 index, address market)
+        internal
+        returns (uint256)
+    {
+        require(index <= orders.length, 'TDR: Out of bounds access');
+
+        IDex dex = IDex(market);
```

Comment:
Is this the address of the Tracer market? If so, how does it work? I thought it would only work if casting from an instance of IDex interface, whereas Tracer is ITracer.

Reply:
Yeah it is. Tracer also implements the IDex interface because of the line `Tracer is SimpleDex`. In this way, the SimpleDex interface is actually implemeneted by Tracer itself

Reply:
Ah true, I didn't realise.